src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -27,8 +27,22 @@ object GoogleExtensions {
 }
 
 class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: AccessPolicyDAO, val googleDirectoryDAO: GoogleDirectoryDAO, val googlePubSubDAO: GooglePubSubDAO, val googleIamDAO: GoogleIamDAO, val googleStorageDAO: GoogleStorageDAO, val googleKeyCache: GoogleKeyCache, val googleServicesConfig: GoogleServicesConfig, val petServiceAccountConfig: PetServiceAccountConfig, extensionResourceType: ResourceType)(implicit val executionContext: ExecutionContext) extends LazyLogging with FutureSupport with CloudExtensions {
+
+  @deprecated
   private[google] def toProxyFromUser(subjectId: WorkbenchUserId): WorkbenchEmail = WorkbenchEmail(s"PROXY_${subjectId.value}@${googleServicesConfig.appsDomain}")
 
+  private val maxGroupEmailLength = 64
+
+  private[google] def toProxyFromUser(user: WorkbenchUser): WorkbenchEmail = {
+    val maxUsernameLength = maxGroupEmailLength -
+      "_".length -
+      user.id.value.length -
+      "@".length -
+      googleServicesConfig.appsDomain.length
+    val username = user.email.value.split("@").head
+    WorkbenchEmail(s"${username.take(maxUsernameLength)}_${user.id.value}@${googleServicesConfig.appsDomain}")
```

Comment:
how about
```
val emailSuffix = s"_${user.id.value}@${googleServicesConfig.appsDomain}"
val username = user.email.value.split("@").head
val maxUsernameLength = maxGroupEmailLength - emailSuffix.length
WorkbenchEmail(username.take(maxUsernameLength) + emailSuffix)
```

Reply:
I like that this builds in the "_" and "@" into the length calculation automatically. I'll incorporate that. Thanks.